src/frontend/js/components/UserMenu/DesktopUserMenu.tsx

```diff
@@ -0,0 +1,44 @@
+import React from 'react';
+import { useIntl } from 'react-intl';
+import { useSelect } from 'downshift';
+import { UserMenuProps, userMenuMessages } from '.';
+
+const DesktopUserMenu: React.FC<UserMenuProps> = ({ user, links }) => {
+  const labels = links.map((label) => label);
+  const intl = useIntl();
+  const { isOpen, highlightedIndex, getMenuProps, getItemProps, getToggleButtonProps } = useSelect({
+    items: labels,
+  });
+
+  return (
+    <div className="user-menu user-menu--desktop selector">
+      <button {...getToggleButtonProps()} className="selector__button">
+        {user}
+        <svg role="img" className="selector__button__icon" aria-hidden="true">
+          <use xlinkHref="#icon-chevron-down" />
+        </svg>
+      </button>
+      <ul
+        {...getMenuProps()}
+        className={`selector__list ${isOpen ? '' : 'selector__list--is-closed'}`}
+      >
+        {isOpen &&
+          links.map((link, index) => (
+            <li key={`user-link-${link.label}-${index}`} {...getItemProps({ item: link, index })}>
+              <a
+                className={`selector__list__link
+                ${highlightedIndex === index ? 'selector__list__link--highlighted' : ''}`}
+                href={link.href}
+                title={`${intl.formatMessage(userMenuMessages.goTo, { label: link.label })}`}
+              >
+                {link.label}
+              </a>
+            </li>
+          ))}
+      </ul>
+    </div>
+  );
+};
+
+DesktopUserMenu.displayName = 'DesktopUserMenu';
+export default DesktopUserMenu;
```

Comment:
Ok. I use it because I think it's easier to understand the main purpose of a script when we see directly what is export by default. And in the case of component, I like the import syntax: default is the component, the rest is extra stuff like types, etc.

```js
import componentA, { interfaceA } from './componentA'
```